code/go/0chain.net/smartcontract/storagesc/stakepool.go

```diff
@@ -259,8 +259,8 @@ func (sp *stakePool) slash(
 		}
 		edbSlash.DelegatePenalties[id] = dpSlash
 	}
-	// todo we should slash from stake pools not rewards. 0chain issue 1495
-	if err := edbSlash.Emit(event.TagStakePoolReward, balances); err != nil {
+	//Added New Tag for StakePoolPenalty
+	if err := edbSlash.Emit(event.TagStakePoolPenalty, balances); err != nil {
```

Comment:
is stake pool reduced for this amount? I don't see the reduction. At first we should reduce the stakepool and all the delegate pools proportionally in the MPT, and only after it update the events_db. looks like with current implementation events_db will be changed, but MPT won't and balances will be different in different storages

Reply:
here is the code 
```go
		if balance, err := currency.MinusCoin(dp.Balance, dpSlash); err != nil {
			return 0, err
		} else {
			dp.Balance = balance
		}
```